### PR TITLE
Moving 'mesh expansion' to helm, single gateway

### PR DESCRIPTION
**Pull request by costinm**
Will create a single ILB IP address instead of 4 - and use envoy to forward to pilot/mixer/ca.

Note that I created a DNS-over-TCP gateway as well - in the past we had a separate 
ILB of type UDP, in kube-system namespace. I haven't tested yet dnsmasq changes - worse 
case we'll still require a ILB for kube-dns (for mesh expansion to raw VMs - multi-cluster will use the local 
kube-dns and replicate the entries ).

The IP allocated by the ILB can be used in the remote clusters as pilot, mixer, ca address.

### CODE REVIEW

File: install/kubernetes/helm/istio/charts/meshexpansion/templates/config-envoy.yaml [hunk 1]
```diff
@@ -0,0 +1,181 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: meshexpansion-envoy-config
+  namespace: istio-system
```
Comment (rshriram) [inline install/kubernetes/helm/istio/charts/meshexpansion/templates/config-envoy.yaml]: Why can’t we do the same thing as the gateways ? The istio-system namespace can be dynamically obtained from helm stuff

File: install/kubernetes/helm/istio/charts/meshexpansion/templates/config-envoy.yaml [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+        filter_chains:
+        - filters:
+          - name: envoy.tcp_proxy
+            config:
+              deprecated_v1: true
+              value:
+                stat_prefix: tcp_pilot
+                route_config:
+                    routes:
+                    - cluster: zvpn.pilot
```
Comment (rshriram) [inline install/kubernetes/helm/istio/charts/meshexpansion/templates/config-envoy.yaml]: You don’t need deprecated v1, if it’s just a single cluster with no selection. Use the cluster parameter in v2 tcp filter

File: install/kubernetes/helm/istio/charts/meshexpansion/templates/config-envoy.yaml [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+        lb_policy: ROUND_ROBIN
+        hosts:
+          - { socket_address: { address: "10.0.0.10", port_value: 53 }}
+
+      # Connection to Mixer. Because policy and telemetry share the same port, currently only
+      # telemetry is supported. Policy should be deployed locally.
+      - name: zvpn.mixer
+        connect_timeout: 5s
+        lb_policy: ROUND_ROBIN
+        type: EDS
```
Comment (rshriram) [inline install/kubernetes/helm/istio/charts/meshexpansion/templates/config-envoy.yaml]: Why eds? Use dns here as well. More portable to have out of kube mixer

File: install/kubernetes/helm/istio/values-istiotest.yaml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+     cpu: 1600m
+     memory: 4Gi
+
+ingress:
+  resources:
+    limits:
+     cpu: 2000m
+     memory: 500Mi
+    requests:
+     cpu: 2000m
```
Comment (rshriram) [inline install/kubernetes/helm/istio/values-istiotest.yaml]: May be call this file meshexpansion-test

File: install/kubernetes/helm/istio/values-istiotest.yaml [hunk 5]
```diff
@@ -0,0 +1,62 @@
+# Configuration overrides for deploying istio for testing
+
+global:
+  nodePort: true
+  imagePullPolicy: Always
+  proxy:
+    image: proxy_debug
```
Comment (rshriram) [inline install/kubernetes/helm/istio/values-istiotest.yaml]: Proxyv2?

File: install/kubernetes/helm/istio/charts/meshexpansion/templates/config-envoy.yaml (hunk 1 above)
Comment (costinm) [reply install/kubernetes/helm/istio/charts/meshexpansion/templates/config-envoy.yaml]: Done

File: install/kubernetes/helm/istio/charts/meshexpansion/templates/service.yaml [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  ports:
+  - name: grpc-ca-mtls
+    port: 8060
+  - name: grpc-pilot
+    port: 15010
+  - name: https-grpc-pilot
+    port: 15011
+  - name: grpc-mixer-mtls
+    port: 15006
+  - port: 15053
```
Comment (sdake) [inline install/kubernetes/helm/istio/charts/meshexpansion/templates/service.yaml]: style issue - a little weird :)

File: install/kubernetes/helm/istio/values.yaml [hunk 7]
```diff
@@ -399,3 +401,8 @@ zipkin:
     # requests:
     #  cpu: 100m
     #  memory: 128Mi
+
+# Meshexpansion adds internal load balancers and gateway allowing standalone machines and multiple
+# clusters to connect to the control plane running in the deployed cluster.
+meshexpansion:
+  enabled: true # For testing
```
Comment (sdake) [inline install/kubernetes/helm/istio/values.yaml]: nice!

Comment (sdake) [reply install/kubernetes/helm/istio/values.yaml]: I just noticed this wasn't in a testing file, but the main values.yaml.  Was the intent to force enable mesh expansion for 0.8?

**Comment by costinm** [comment]
Closing, will use ingress gateway instead.

**Pull request closed without merging**